### ISSUE TITLE
Datagrid: Select Rows Across Pages not working with filter

### ISSUE THREAD

**Issue by tmcconechy** [infor-design/enterprise#3601]
**Describe the bug**
When using selectRowsAcrossPages setting the selected rows can be reset by filtering.

**To Reproduce**
1. change the code of test-paging-multiselect-select-across-page to below
1. Go to http://localhost:4000/components/datagrid/test-paging-multiselect-select-across-page.html
2. Select two rows on page one
3. Filter on 214229 (values on page two)
4. Select one or just go back to page one 

**Expected behavior**
Ideally the values can be saved across filter

**Version**
- ids-enterprise: [e.g. v4.26]

**Platform**
 - Chrome/Mac (any)

**Additional context**
Ms teams chat

**New Code **
Just added filter options to the example to test:
```

<div class="row">
  <div class="twelve columns">
    <div role="toolbar" class="toolbar">

      <div class="title">
        Data Grid Header Title
        <span class="datagrid-result-count">(N Results)</span>
      </div>

      <div class="buttonset">

      </div>

      <div class="more">
        <button class="btn-actions" type="button">
          <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
            <use xlink:href="#icon-more"></use>
          </svg>
          <span class="audible">More Actions</span>
        </button>
        <ul class="popupmenu">
          <li><a href="#">Option One</a></li>
          <li><a href="#">Option Two</a></li>
          <li><a href="#">Option Three</a></li>
          <li class="separator single-selectable-section"></li>
          <li class="heading">Row Height</li>
          <li class="is-selectable"><a data-option="row-short" href="#">Short</a></li>
          <li class="is-selectable"><a data-option="row-medium" href="#">Medium</a></li>
          <li class="is-selectable is-checked"><a data-option="row-normal" href="#">Normal</a></li>
        </ul>

      </div>
    </div>

    <div class="contextual-toolbar toolbar is-hidden">
      <div class="title selection-count">1 Selected</div>
      <div class="buttonset">
        <button class="btn-icon" type="button" id="remove-btn">
          <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
            <use xlink:href="#icon-delete"></use>
          </svg>
          <span class="audible">Remove</span>
        </button>
      </div>
    </div>

    <div id="datagrid">

    </div>
  </div>
</div>

<script>
  $('body').one('initialized', function () {

      var grid,
        columns = [],
        data = [];

      //Define Columns for the Grid.
      columns.push({ id: 'selectionCheckbox', sortable: false, resizable: false, formatter: Formatters.SelectionCheckbox, align: 'center'});
      columns.push({ id: 'id', name: 'Id', field: 'id'});
      columns.push({ id: 'productId', name: 'Product Id', field: 'productId', width: 140, filterType: 'text'});
      columns.push({ id: 'productName', name: 'Product Name', sortable: false, field: 'productName', width: 150, formatter: Formatters.Hyperlink, editor: Editors.Input, filterType: 'text'});
      columns.push({ id: 'activity', hidden: true, name: 'Activity', field: 'activity', width: 125, editor: Editors.Input, filterType: 'text'});
      columns.push({ id: 'quantity', name: 'Quantity', field: 'quantity', width: 125});
      columns.push({ id: 'price', name: 'Price', field: 'price', width: 125, formatter: Formatters.Decimal});
      columns.push({ id: 'orderDate', name: 'Order Date', field: 'orderDate', formatter: Formatters.Date, dateFormat: 'M/d/yyyy'});

      //Init and get the api for the grid
      grid = $('#datagrid').datagrid({
        columns: columns,
        selectable: 'multiple',
        paging: true,
        filterable: true,
        editable: false,
        selectable: 'multiple',
        allowSelectAcrossPages: true,
        pagesize: 5,
        pagesizes: [5, 10, 15, 20],
        source: function(req, response) {
          var url = '{{basepath}}api/compressors?pageNum='+ req.activePage +'&pageSize='+ req.pagesize;

          if (req.sortField) {
            url += '&sortField=' + req.sortField + '&sortAsc=' + req.sortAsc;
          }

          if (req.filterExpr && req.filterExpr[0]) {
            url += '&filterValue=' + req.filterExpr[0].value;
            url += '&filterOp=' + req.filterExpr[0].operator;
            url += '&filterColumn=' + req.filterExpr[0].columnId;
          }

          //Get Page Based on info in Req, return results into response;
          $.getJSON(url, function(res) {
            req.total = res.total;
            req.preserveSelected = true;
            response(res.data, req);
          });
        },
        toolbar: {title: 'Data Grid Header Title', results: true, dateFilter: false ,keywordFilter: true, actions: true, views: true, rowHeight: true, collapsibleFilter: true}
      }).on('selected', function (e, args, action) {
        console.log(action, args);
      });

  });
</script>

```

**Comment by dhumalkishor** [comment]
Hi Tim,
As per your comments you are going to address below issue along with this issue.
https://teams.microsoft.com/l/message/19:abd8955de12846b4aaaf10f108f2f36e@thread.skype/1581431786827?tenantId=457d5685-0467-4d05-b23b-8f817adda47c&groupId=4f50ef7d-e88d-4ccb-98ca-65f26e57fe35&parentMessageId=1581431786827&teamName=IDS%20Enterprise%20Development%20(SoHo%20Xi)&channelName=Breaking%20Changes&createdTime=1581431786827

**Comment by tmcconechy** [comment]
Ok, that issue is actually this one - https://github.com/infor-design/enterprise-ng/issues/588
We are trying to fix that one currently. But may need to pull it out. This issue is around filter.

This issue is a bit harder - we shall see if we can address this too.

**Comment by davidJayaraj** [comment]
Hello Tim,

Do you have anytime to get this issue resolved. As we have few customer waiting for this fix and is critical for us. 

Thanks,
David Jayaraj

**Comment by tmcconechy** [comment]
@davidJayaraj we have this in the current sprint. See above a pull request has been raised. If you would like to test the pull request please do. Assuming it solves the problem this will be in our early July next release.

**Comment by dhumalkishor** [comment]
Hi Tim, Can we get this in 4.29.0 version??

**Comment by tmcconechy** [comment]
Bit too late for that 4.29 is already in RC. This needs to go through full QA

**Comment by tmcconechy** [comment]
Also as i workaround i can suggest not doing this. As its technically a feature. You can do an ajax call for all data and then just use clientSide paging.

**Comment by janahintal** [comment]
QA Failed. 

Scenario 1: https://master-enterprise.demo.design.infor.com/components/datagrid/test-paging-multiselect-select-across-page.html
1. Select a row
2. Filter 214229 on Product Id
3. Remove filter by clear the filter value 214229
4. Selected row should remain selected

![image](https://user-images.githubusercontent.com/45757016/84026043-2bfbee80-a9bf-11ea-9f64-ebdefd5800c9.png)

Scenario 2: https://master-enterprise.demo.design.infor.com/components/datagrid/test-paging-multiselect-select-across-page.html

1. Go to 2nd or after page
2. Filter 214229 on Product Id
3. There should be one row in result for grid table

![image](https://user-images.githubusercontent.com/45757016/84026721-687c1a00-a9c0-11ea-95d5-a02a169c1eba.png)

Scenario 3: https://master-enterprise.demo.design.infor.com/components/datagrid/test-paging-multiselect-select-across-page.html

1. Select a row and filter 214229 on Product Id
2. Select the filter row while filtering
3. Remove filter by clear the filter value 214229
4. Selected rows should sync properly

![image](https://user-images.githubusercontent.com/45757016/84027290-5058ca80-a9c1-11ea-97e6-59107e6f47e1.png)

**Comment by tmcconechy** [comment]
Master was not correctly updated. Please retest

**Comment by jbrcna** [comment]
**QA Passed**
All scenarios were now working as expected.
v4.30.0-dev
https://master-enterprise.demo.design.infor.com/components/datagrid/test-paging-multiselect-select-across-page.html

**Comment by dhumalkishor** [comment]
Hi Tim,

Please check below link on team, I thik this scenario is not fixed in 4.31.1 soho library.
https://teams.microsoft.com/l/message/19:abd8955de12846b4aaaf10f108f2f36e@thread.skype/1585568141613?tenantId=457d5685-0467-4d05-b23b-8f817adda47c&groupId=4f50ef7d-e88d-4ccb-98ca-65f26e57fe35&parentMessageId=1581431786827&teamName=IDS%20Enterprise%20Development%20(SoHo%20Xi)&channelName=Breaking%20Changes&createdTime=1585568141613

**Comment by tmcconechy** [comment]
This has passed as tested. Can you make a new issue with the steps against the latest examples as to what your issue/enhancement is related to this.
